src/utils/fs.h

```diff
@@ -38,15 +39,51 @@
 #ifndef UTILS_FS_H_
 #define UTILS_FS_H_
 
+
+
+#ifdef _WIN32
+#include <windows.h>
+#define MAX_PATH_SIZE MAX_PATH
+#endif
+
+#if defined(__linux__) || defined(__APPLE__)
+#include <limits.h>
+#define MAX_PATH_SIZE (PATH_MAX + 1)
+#endif
+
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @param path buffer with size MAX_PATH_SIZE where function stores path to executable
+ * @return 1 - SUCCESS, 0 - ERROR
+ */ 
+int get_exec_path(char* path);
+
+
 const char *get_temp_dir(void);
 
 #ifdef __cplusplus
+} // extern "C"
+#endif
+
+
+
+#ifdef __cplusplus
+#include <string>
+
+inline std::string get_executable_path(){
+        std::string path(MAX_PATH_SIZE, '\0');
+        if (!get_exec_path(path.data())){
+            return "";
+        }
+        path.erase(path.find('\0'));
```

Comment:
Assuming '\0' is always present seems a bit tricky to me, however it should hold. Perhaps using iterator-based find could be better, as erase would raise exception if find returns string::npos.